keploy/keploy.go

```diff
@@ -383,20 +410,79 @@ func (k *Keploy) simulate(tc models.TestCase) (*models.HttpResp, error) {
 	return &resp.Resp, nil
 }
 
-func (k *Keploy) check(runId string, tc models.TestCase) bool {
-	resp, err := k.simulate(tc)
+func (k *Keploy) simulateGrpc(tc models.TestCase) (models.GrpcResp, error) {
+	// add dependencies to shared context
+	k.deps.Store(tc.ID, tc.Deps)
+	defer k.deps.Delete(tc.ID)
+	// add mocks to shared context
+	k.mocks.Store(tc.ID, tc.Mocks)
+	defer k.mocks.Delete(tc.ID)
+	k.mocktime.Store(tc.ID, tc.Captured)
+	defer k.mocktime.Delete(tc.ID)
+	tid := string(tc.ID)
+	port := k.cfg.App.Port
+	if port[0] != ':' {
+		port = ":" + port
+	}
+	m := sync.Mutex{}
+	m.Lock()
+	k.PutRespGrpc(tc.ID, GrpcResp{L: &m})
+
+	// The simulate call is done via grpcurl which acts as a grpc client
+	err := GrpCurl(tc.GrpcReq.Body, `tid:`+tid, "localhost"+port, tc.GrpcReq.Method)
 	if err != nil {
-		k.Log.Error("failed to simulate request on local server", zap.Error(err))
-		return false
+		k.Log.Error("failed to simulate grpc request", zap.String("testcase id:", tc.ID), zap.Error(err))
+	}
+	m.Lock()
+	m.Unlock()
+	resp := k.GetRespGrpc(tc.ID)
+	defer k.resp.Delete(tc.ID)
+	return resp.Resp, nil
+}
+
+func (k *Keploy) check(runId string, tc models.TestCase) bool {
+	var (
+		resp     *models.HttpResp
+		respGrpc models.GrpcResp
+		bin      []byte
+		err      error
+	)
+	switch tc.Type {
+	case string(models.HTTP):
```

Comment:
Since testcase.Type is a newly introduced field, may I check if this will cause any backward incompatibility issue for existing test cases that were recorded before this change, which will contain empty string on this field?

Reply:
Hi **@jiawchong**, These changes are compatible for previously generated yaml test-cases because this uses "kind" field of previous yaml test-cases for `tc.Type`. 
But if you are using mongo instead of yaml files, then you may have issues. If you are facing issues, please report it.